### ISSUE TITLE
no-empty-interface should not fail on an interface that resolves types

### ISSUE THREAD

**Issue by bronson** [palantir/tslint#3256]
### Bug Report

Related to https://github.com/palantir/tslint/issues/2065

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.2
- __Running TSLint via__: Atom

#### TypeScript code being linted

```ts
interface CustomPromise extends Promise<string|false> {
}
```
This throws the error, "An interface declaring no members is equivalent to its supertype".

That's not really true, is it?  Here, CustomPromise is not equivalent to Promise.

**Comment by andy-hanson** [comment]
Why not `type CustomPromise = Promise<string | false>`?

**Comment by ajafff** [comment]
I guess that's because you cannot extend a type alias.
